### PR TITLE
DEVX-1826: Updates python example to Serialization API

### PR DESCRIPTION
**Pull request by rspurgeon**

### CODE REVIEW

File: clients/cloud/python/consumer.py
Comment (rspurgeon) [inline clients/cloud/python/consumer.py]: I'm not sure about this change.  On my system, where I have a python 2.7 installation, `env python` resolves to python 2, and cannot locate the confluent-kafka package or run the examples.  I'm not sure of the recommended method for making this work on a system like mine, or if it should be a system specific configuration change instead of this.  @rnpridgeon can you help me if this is a valid change or should users be doing something local that I can document?

File: clients/cloud/python/README.md
Comment (ybyzek) [inline clients/cloud/python/README.md]: Python2 is no longer supported, so propose

```suggestion
* [Confluent's Python Client for Apache Kafka](https://github.com/confluentinc/confluent-kafka-python) installed on your machine. Check that you are using version 1.4.2 or higher (e.g., `pip3 show confluent-kafka`).
```

File: clients/cloud/python/consumer.py
Comment (ybyzek) [reply clients/cloud/python/consumer.py]: I defer to @rnpridgeon but I believe users  often alias `python` to their local `python2` or `python3`, so consider leaving as `python`.

**Review by ybyzek — APPROVED** [review]
LGTM pending approval from @confluentinc/clients

**Comment by ybyzek** [comment]
@rspurgeon did you validate the Docker-based Python example and determine whether https://github.com/confluentinc/examples/blob/5.5.0-post/clients/cloud/python/requirements.txt also needs an update?

File: clients/cloud/python/consumer.py
Comment (rspurgeon) [reply clients/cloud/python/consumer.py]: IIUC `env` does not reference aliases

File: clients/cloud/python/README.md
Comment (rspurgeon) [reply clients/cloud/python/README.md]: This is similar to the `python` shebang issue above.  I'm not familiar enough with the python ecosystem.  If we change the env above, I think this makes sense stating that we only support python3, but if we don't change above, then this seems confusing.

File: clients/cloud/python/consumer.py
Comment (rspurgeon) [reply clients/cloud/python/consumer.py]: In my case I used pip3 (aliased on my machine from pip to pip3) to install confluent-kafka, but python 2 is found first in my path. I think this causes the issue I see when trying to run the scripts directly.  I assume others might experience this, so outside of uninstalling python2, adjusting my PATH order, or also using pip 2 to install confluent-kafka, is there a better remedy?

**Comment by rspurgeon** [comment]
> @rspurgeon did you validate the Docker-based Python example and determine whether https://github.com/confluentinc/examples/blob/5.5.0-post/clients/cloud/python/requirements.txt also needs an update?

@ybyzek the Docker instructions work.   When building the docker image the following warnings are given, but the applications work as expected:

```
WARNING: confluent-kafka 1.4.2 does not provide the extra 'jsonschema'
...
WARNING: You are using pip version 20.0.2; however, version 20.1.1 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command
```

The addition of `jsonschema` was made by @javabrett in https://github.com/confluentinc/examples/pull/603

File: clients/cloud/python/README.md
Comment (rnpridgeon) [reply clients/cloud/python/README.md]: Virtualenv should be used rather than relying on the system installation. This will simplify things as you can explicitly instruct it to run a particular interpreter. 

https://virtualenv.pypa.io/en/latest/user_guide.html#introduction

File: clients/cloud/python/consumer.py
Comment (rnpridgeon) [reply clients/cloud/python/consumer.py]: see comment about virtualenv

File: clients/cloud/python/README.md
Comment (rspurgeon) [reply clients/cloud/python/README.md]: 🤦 , of course... Thanks @rnpridgeon .  I'm going to add some documentation to help users if this is new to them or have memory failure like I do.

Comment (rspurgeon) [reply clients/cloud/python/README.md]: I've added some instructions for installing and running Virtualenv to the README

**Comment by rspurgeon** [comment]
@confluentinc/clients Could someone provide a review for this?  Thank you

File: clients/cloud/python/README.md
Comment (mhowlett) [inline clients/cloud/python/README.md]: the new apis are 'experimental and subject to change', is the main thing. In practice, they're essentially stable, except that I hope they'll be combined with the Producer and Consumer classes (not exposed as separate classes), and the details of that aren't completely clear.

i'm not sure how best to communicate this.

Comment (mhowlett) [reply clients/cloud/python/README.md]: Probably don't want/need the `#` in the url.

File: clients/cloud/python/ccloud_lib.py
Comment (mhowlett) [inline clients/cloud/python/ccloud_lib.py:68]: is there a reason to change the literal to a dict constructor? I'd use a literal and put it on one line `return { "name": self.name }`. but kinda bike shedding.

File: clients/cloud/python/ccloud_lib.py
Comment (mhowlett) [inline clients/cloud/python/ccloud_lib.py:111]: same same

File: clients/cloud/python/ccloud_lib.py
Comment (rspurgeon) [reply clients/cloud/python/ccloud_lib.py:68]: I modeled these changes after the official examples: https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/avro_consumer.py#L62

**Review by mhowlett — APPROVED** [review]
LGTM, except i'm unsure how or if to communicate the API is marked as experimental.

**Comment by rspurgeon** [comment]
> LGTM, except i'm unsure how or if to communicate the API is marked as experimental.

Thanks @mhowlett , I'm going to update the URL and push that minor change.   Regading the API stability, maybe if the official docs are updated with verbiage, we can copy and/or point to that. https://docs.confluent.io/current/clients/confluent-kafka-python/ (no mention of experimental Serialization API, at least on this front page).

File: clients/cloud/python/ccloud_lib.py
Comment (ybyzek) [inline clients/cloud/python/ccloud_lib.py:142]: Thank you!  ref: https://github.com/confluentinc/examples/issues/661#issuecomment-640745841

**Pull request merged**
